tools/serverpod_cli/lib/src/migrations/generator.dart

```diff
@@ -122,19 +140,34 @@ class MigrationGenerator {
     required bool force,
     required int priority,
     bool write = true,
+    required GeneratorConfig config,
   }) async {
     var migrationRegistry = MigrationRegistry.load(
       migrationsProjectDirectory,
     );
 
+    var protocols =
+        await ProtocolHelper.loadProjectYamlProtocolsFromDisk(config);
+    var entityDefinitions = StatefulAnalyzer(protocols).validateAll();
```

Comment:
Suggestion: Print warnings and errors and stop if any are found.

This should be found in a e2e test as well. But it looks like the printing and stopping has been removed this was moved from old class.